packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,872 @@
+
+  /**
+   * Specifies a standard filter (true) or no filter (false) for continuous logging.
+   * Choosing the standard filter prunes out non-useful Apache Spark driver/executor and Apache Hadoop YARN heartbeat log messages.
+   * Choosing no filter gives all the log messages.
+   */
+  ENABLE_CONTINUOUS_LOG_FILTER = '--enable-continuous-log-filter',
+
+  /**
+   * Specifies a custom Amazon CloudWatch log group name for a job enabled for continuous logging.
+   */
+  LOG_GROUP = '--continuous-log-logGroup',
+
+  /**
+   * Specifies a custom CloudWatch log stream prefix for a job enabled for continuous logging.
+   */
+  LOG_STREAM_PREFIX = '--continuous-log-logStreamPrefix',
+
+  /**
+   * Specifies a custom conversion log pattern for a job enabled for continuous logging.
+   */
+  LOG_CONVERSION_PATTERN = '--continuous-log-conversionPattern',
+
+  /**
+   * Enables Apache Spark web UI.
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+   */
+  ENABLE_SPARK_UI = '--enable-spark-ui',
+
+  /**
+   * Specifies the Amazon S3 path for storing the Spark event logs for the job.
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+   */
+  SPARK_UI_LOGS_PATH = '--spark-event-logs-path',
+}
+
+/**
+ * Interface representing a created or an imported {@link Job}.
+ */
+export interface IJob extends cdk.IResource {
+  /**
+   * The name of the job.
+   * @attribute
+   */
+  readonly jobName: string;
+
+  /**
+   * The ARN of the job.
+   * @attribute
+   */
+  readonly jobArn: string;
+
+  /**
+   * Defines a CloudWatch event rule triggered when something happens with this job.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#glue-event-types
+   */
+  onEvent(id: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+   * Defines a CloudWatch event rule triggered when this job moves to the SUCCEEDED state.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#glue-event-types
+   */
+  onSuccess(id?: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+   * Defines a CloudWatch event rule triggered when this job moves to the FAILED state.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#glue-event-types
+   */
+  onFailure(id?: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+   * Defines a CloudWatch event rule triggered when this job moves to the TIMEOUT state.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#glue-event-types
+   */
+  onTimeout(id?: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+   * Create a CloudWatch metric.
+   *
+   * @param metricName name of the metric typically prefixed with `glue.driver.`, `glue.<executorId>.` or `glue.ALL.`.
+   * @param jobRunId a dimension that filters for metrics of a specific JobRun ID, or `ALL`.
```

Comment:
don't require for this, add a separate `metricAll` method

Reply:
I'm not sure I'm following? Are you saying break it into 2 methods
1. One with the `jobRunId` param and call it `metric`
2. One with no `jobRunId` param and call it `metricAll` and under the hood set the `jobRunId` to `ALL`?

Reply:
This is why I shouldn't review code late at night :) I mis-read this, didn't see the "run" part. In my opinion, there's very little reason why a user would specify a specific job run ID here. The lifecycle for the CDK application is much longer than individual job runs, meaning there's a very small chance that metrics for a specific run are at all relevant after deployment. To that end, let's go just for your option 2 above:
```ts
metric(metricName: string, type: MetricType, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
```

Reply:
What you say makes sense, as such I'll be updating this.